node-graph/gstd/src/gpu_nodes.rs

```diff
@@ -442,7 +442,22 @@ async fn blend_gpu_image(foreground: ImageFrame<Color>, background: ImageFrame<C
 		..Default::default()
 	};
 	log::debug!("compiling network");
-	let proto_networks = compiler.compile(network.clone()).collect();
+
+	let proto_networks_result = compiler.compile(network.clone());
+
+	if proto_networks_result.is_err() {
+		return ImageFrame {
+			image: Image {
+				data: Vec::<Color>::new(),
+				width: 0,
+				height: 0,
+			},
+			transform: DAffine2::ZERO,
+		};
+	}
+
+	let proto_networks = proto_networks_result.unwrap().collect();
```

Comment:
You can use a let else block here instead
```rust
let Ok(proto_networks) = compiler.compile(network.clone() else {return ImageFrame::empty()};
```

Reply:
Thanks, that's super helpful. I think we still need to call `collect()` on the result so I have done something along the lines of the below, which seems to work:

```
let Ok(proto_networks_result) = compiler.compile(network.clone()) else {return ImageFrame::empty()};
let proto_networks = proto_networks_result.collect();
```